R/derive_param_bor.R

```diff
@@ -0,0 +1,518 @@
+#' Adds a Parameter for the Best Overall Response (without confirmation)
+#
+#' @description
+#'     Derives Best Overall Response (optionally) up to first Progressive Disease
+#
+#' @details
+#'    Calculates the best overall response (BOR) parameter, as detailed below.
+#'
+#'    Records after PD can be removed using the source_pd and source_datasets
+#'    arguments.
+#'
+#'   Note:
+#'   1. All `CR`, `PR` and `PD` response records are considered for Best Overall Response.
+#'
+#'   2. All `SD` or `NON-CR/NON-PD` records where `ADT` >= `reference_date` +
+#'      `ref_start_window` are also considered for Best Overall Response.
+#'
+#'   3. Subjects with **ONLY** an `SD` or `NON-CR/NON-PD` records where `ADT` <
+#'      `reference_date` + `ref_start_window` are assigned a Best Overall Response of `NE`.
+#'
+#'   4. The Best Response, by `subject_keys` of the records in steps 1 to 3, is then
+#'      selected in the following order of preference:
+#'      ```{r, eval=FALSE}
+#'                  AVALC %in% c("CR") ~ 1,
+#'                  AVALC %in% c("PR") ~ 2,
+#'                  AVALC %in% c("SD") ~ 3,
+#'                  AVALC %in% c("NON-CR/NON-PD") ~ 4,
+#'                  AVALC %in% c("PD") ~ 5,
+#'                  AVALC %in% c("NE") ~ 6,
+#'                  is.null(AVALC) ~ 7)) %>%
+#'      ```
+#'   5. The `AVAL` column is added and set using the `aval_fun(AVALC)` function
+#'
+#'   6. The columns specified by the `set_values_to` parameter and records
+#'      are added to the dataframe passed into the `dataset` argument
+#'
+#'  Also Note: All columns from the input dataset are kept. For subjects with no records in
+#'  the input dataset (after the filter is applied) all columns are kept from ADSL which are
+#'  also in the input dataset.  Columns which are not to be populated for the new parameter
+#'  or populated differently (e.g. RSSTRESC, VISIT, PARCATy, ANLzzFL, ...) should be
```

Comment:
Please use backticks for variable names like `RSSTRESC`.

Reply:
Updated